### PR TITLE
gpu: sycl: sum: implemented

### PR DESCRIPTION
**Pull request by t4c1**
Add SYCL implementation for sum primitive. There is one implementation that handles up to 16 inputs and another one that repeatedly uses the first one for more inputs.

co-authored with @kala855

### CODE REVIEW

**Comment by t4c1** [comment]
Is the CI timing out on Win Server 2022 expected? I have the same happening on another PR after rebasing on main, just like on this one.

**Comment by vpirogov** [comment]
@t4c1, these timeouts are expected. Looks like MSVC update from 19.39.33523.0 to 19.40.33811.0 in Azure environment breaks `gtest` framework.

**Review by kala855 — APPROVED** [review]

File: src/xpu/sycl/types.hpp [hunk 1]
```diff
@@ -30,6 +30,21 @@ namespace impl {
...
+#define DNNL_ARG_SRC_8 9
+#define DNNL_ARG_SRC_9 10
+#define DNNL_ARG_SRC_10 11
+#define DNNL_ARG_SRC_11 12
+#define DNNL_ARG_SRC_12 13
+#define DNNL_ARG_SRC_13 14
+#define DNNL_ARG_SRC_14 15
+#define DNNL_ARG_SRC_15 16
+
+#define MAX_NUM_TENSORS 16
```
Comment (mgouicem) [inline src/xpu/sycl/types.hpp:46]: Is this specific to sycl sum implementation? If so I would advice to name it accordingly (e.g. DNNL_REF_SUM_MAX_NUM_TENSORS)

Comment (t4c1) [reply src/xpu/sycl/types.hpp]: done

**Comment by t4c1** [comment]
@mgouicem I think I addressed your comments and this PR has been sitting there for months. When can I expect this to get merged?

**Review by mgouicem — APPROVED** [review]

File: src/xpu/sycl/types.hpp [hunk 2]
```diff
@@ -30,6 +30,21 @@ namespace impl {
 namespace xpu {
 namespace sycl {
 
+#define DNNL_ARG_SRC_4 5
```
Comment (densamoilov) [inline src/xpu/sycl/types.hpp:33]: It doesn't look like this code belongs here. Can you move it to `sum_kernels.hpp`?

Comment (t4c1) [reply src/xpu/sycl/types.hpp:33]: done

**Pull request merged**